### PR TITLE
Add global limit

### PR DESCRIPTION
**Pull request by jannyHou**
#### Related issues

https://github.com/strongloop/loopback-connector-couchdb2/issues/41

### Checklist

- [x] New tests added or existing tests modified to cover all changes
- [x] Code conforms with the [style
  guide](http://loopback.io/doc/en/contrib/style-guide.html)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -48,4 +51,4 @@
     "url": "https://github.com/strongloop/loopback-connector-cloudant/issues"
   },
   "license": "Artistic-2.0"
-}
+}
```
Comment (virkt25) [inline package.json]: Need newline

File: package.json [hunk 2]
```diff
@@ -5,6 +5,9 @@
     "tag": "lts"
   },
   "description": "LoopBack Connector for IBM Cloundant DB",
+  "engines": {
+    "node": ">= 4 <= 6"
```
Comment (virkt25) [inline package.json]: Any reason this doesn't work with 8?

File: test/count.test.js [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+// License text available at https://opensource.org/licenses/Apache-2.0
+
+'use strict';
+
+var _ = require('lodash');
+var should = require('should');
+var COUNT_OF_SAMPLES = 70;
+var db, TestCountUser;
+
+function create50Samples() {
```
Comment (virkt25) [inline test/count.test.js]: Please rename the function to just `createSamples`. It's misleading right now as it creates 70 samples but is called `create50Samples`

File: lib/cloudant.js [hunk 4]
```diff
@@ -784,6 +786,15 @@ Cloudant.prototype.automigrate = function(models, cb) {
  *        "<key2>": -1
  *
 */
+
+Cloudant.prototype.getLimit = function(limit) {
+  return limit || this.getGlobalLimit();
+};
+
+Cloudant.prototype.getGlobalLimit = function() {
```
Comment (virkt25) [inline lib/cloudant.js]: Any reason for this to be a function and not just be a part of `getLimit`? 

```
Cloudant.prototype.getLimit = function(limit) {
  return limit || this.settings.globalLimit;
}
```

File: test/id.test.js [hunk 5]
```diff
@@ -18,8 +18,7 @@ describe('using _id as id', function() {
       owner: {type: String},
       _id: {type: String, id: true},
     }, {forceId: true});
-
-    db.automigrate('Task', done);
```
Comment (virkt25) [inline test/id.test.js]: Is removing the `db.automigrate` intentional? (Seems unrelated to the issue -- just want to confirm it wasn't accidental).

File: package.json (hunk 2 above)
Comment (jannyHou) [reply package.json]: @virkt25 This is branch 1.x and I copied the engines from `loopback-datasource-juggler@2.x`, I think they have the same engines support

File: test/count.test.js (hunk 3 above)
Comment (jannyHou) [reply test/count.test.js]: good catch

File: test/id.test.js (hunk 5 above)
Comment (jannyHou) [reply test/id.test.js]: @virkt25 1.x has a bug with `automigrate` function, which is fixed in 2.x, if you look at all the 1.x tests, they don't call `automigrate`, calling it results in a document conflict error.

File: lib/cloudant.js (hunk 4 above)
Comment (jannyHou) [reply lib/cloudant.js]: It's more like a personal coding habit, I treat `getGlobalLimit` and `getLimit` as separate functionalities, e.g. you can add more approaches set the global limit value, currently it's from datasource settings only, but it may come from model or Env in the future.
And for `getLimit`, we may add options and corresponding logic.

File: test/id.test.js (hunk 5 above)
Comment (jannyHou) [reply test/id.test.js]: find the related PR:
https://github.com/strongloop/loopback-connector-cloudant/commit/e348c2576d7fead66ce7370871fe5cfecef56539
and 
https://github.com/strongloop/loopback-connector-cloudant/pull/109
if you are interested.

File: package.json (hunk 1 above)
Comment (jannyHou) [reply package.json]: turned off `autosave` :(

File: test/id.test.js (hunk 5 above)
Comment (virkt25) [reply test/id.test.js]: Ooh ok. I just wanted to make sure it wasn't accidental and I didn't realize this was for 1.x branch

File: package.json (hunk 2 above)
Comment (virkt25) [reply package.json]: `1.x` can still be used with `3.x` version of Juggler. I disagree with it only supporting what 2.x supports because 3.x does support Node 8. Unless this connector version of 1.x shouldn't be used with Juggler 3.x at all.

File: lib/cloudant.js (hunk 4 above)
Comment (virkt25) [reply lib/cloudant.js]: Fair enough. My personal preference is just having one function :P 

This isn't a blocker.

File: package.json (hunk 2 above)
Comment (jannyHou) [reply package.json]: The code of `1.x` is written when node@8.x was not released, and due to the annoying bug of `automigrate`, I would not recommend new users to use it, `1.x` doesn't have the index and view feature either. That's why I don't want to spend effort on its compatibility of node 8, even it passes all test on node 8, I still don't want to give loopback3 users an impression that this old version is an option for them :(

**Comment by jannyHou** [comment]
@virkt25 thanks for the review, good talk. green now :)
